### ISSUE TITLE
clang error that -fopenmp is unsupported on Mac

### ISSUE THREAD

**Issue by cdeil** [bwinkel/cygrid#5]
I'm getting this error:

```
$ pip install cygrid --user
Collecting cygrid
  Downloading cygrid-0.9.5.tar.gz (582kB)
    100% |████████████████████████████████| 583kB 1.0MB/s 
Requirement already satisfied (use --upgrade to upgrade): setuptools in /Users/deil/Library/Python/3.5/lib/python/site-packages (from cygrid)
Requirement already satisfied (use --upgrade to upgrade): cython>=0.20.2 in /opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages (from cygrid)
Requirement already satisfied (use --upgrade to upgrade): numpy>=1.8 in /opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages (from cygrid)
Requirement already satisfied (use --upgrade to upgrade): astropy>=1.0 in /Users/deil/Library/Python/3.5/lib/python/site-packages/astropy-1.2.dev15642-py3.5-macosx-10.11-x86_64.egg (from cygrid)
Installing collected packages: cygrid
  Running setup.py install for cygrid ... error
    Complete output from command /opt/local/Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-n2qwoo50/cygrid/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-qqnodoyd-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.11-x86_64-3.5
    creating build/lib.macosx-10.11-x86_64-3.5/cygrid
    copying cygrid/__init__.py -> build/lib.macosx-10.11-x86_64-3.5/cygrid
    running build_ext
    cythoning cygrid/kernels.pyx to cygrid/kernels.cpp
    building 'cygrid.kernels' extension
    creating build/temp.macosx-10.11-x86_64-3.5
    creating build/temp.macosx-10.11-x86_64-3.5/cygrid
    /usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -pipe -Os -I/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c cygrid/kernels.cpp -o build/temp.macosx-10.11-x86_64-3.5/cygrid/kernels.o -fopenmp -O3 -std=c++11
    clang: error: unsupported option '-fopenmp'
    error: command '/usr/bin/clang' failed with exit status 1

    ----------------------------------------
Command "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-n2qwoo50/cygrid/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-qqnodoyd-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-n2qwoo50/cygrid/
```

I think most people on Mac will hit this error, i.e. effectively install is currently broken on Mac.

Do you want to support Mac (and Windows)?

If yes, the first step should be to add a Mac build on travis-ci and a Windows build on Appveyor.

I remember some discussion and investigation on how to handle OpenMP compilation here:
https://github.com/astropy/imageutils/pull/35#issuecomment-60656910

The solution they implemented is here:
https://github.com/astropy/astroscrappy/blob/e28afa4d27201df87044625e0e23147778fb70a1/astroscrappy/utils/setup_package.py#L23

It's pretty complex though and uses astropy-helpers, i.e. is a non-trivial change for you and probably a lot of tedious work to get this working properly. Personally I don't really need this package, I just wanted to try it out, so don't feel the need to go down that path on my behalf.

Maybe for now add a comment to the README that this is a known issue on Mac?

**Comment by DanielLenz** [comment]
I'm running OS X on my system and have been able to install cygrid on both python 2.7 and 3.5. However, I'm using the gcc-5 provided by [homebrew](http://brew.sh) instead of clang. I will look into the issue with openmp and clang later, but using gcc instead might be a quick fix for now.
Here's a [link](http://apple.stackexchange.com/questions/212274/cannot-symlink-gcc-5-instead-of-clang-on-os-x-10-10-5) on how to symlink gcc-5 instead of clang.

**Comment by evertrol** [comment]
For me, this failed for the same reason, albeit with a different error:

```
clang++ -bundle -undefined dynamic_lookup build/temp.macosx-10.10-x86_64-3.5/cygrid/kernels.o -L/usr/local/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/sqlite/lib -o build/lib.macosx-10.10-x86_64-3.5/cygrid/kernels.cpython-35m-darwin.so -fopenmp
ld: library not found for -lgomp
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command 'clang++' failed with exit status 1
```

As I'm cautious with symlinking system tools, the following can also do the trick:

```
CFLAGS=-I/usr/local/Cellar/gcc/6.2.0/lib/gcc/6/gcc/x86_64-apple-darwin14.5.0/6.2.0/include \
  CXX=/usr/local/bin/gcc-6 \
  pip install cygrid
```

The horrendous include path is because Homebrew decided not to symlink the include directory. Presumably to keep it out of the way of clang's include directory. Since other people seem to have similar issues getting gcc to find `omp.h` (which would be the next error to get after setting the correct compiler), I'm including this variant as a comment here.

**Comment by bwinkel** [comment]
Thanks a lot! Unfortunately, I'm not having access to any Mac-OS machine. Perhaps, Daniel could look into this in more detail in the near future...

**Comment by DanielLenz** [comment]
I just included a very simple fix in e2b003e, requiring gcc-6.

**Comment by bwinkel** [comment]
Thanks! We should also try to include MacOS into our travis config. Will try to do this later.
